### PR TITLE
test: add comprehensive unit tests for prices utilities

### PR DESCRIPTION
**Pull request by Pankajyadav919**
Description

Adds comprehensive unit tests for prices utilities to ensure correct configuration, safe state handling, and robust behavior across edge cases.

###Covered cases:
createPricesApi correctly initializes with base URL and auth token
Handles missing authToken gracefully (uses undefined)
Handles missing preferences store safely
updatePricesAuthToken updates token correctly
Handles null token updates
Handles missing prices object without runtime errors

###Functions covered
-isConfigured(...)
-createPricesApi(...)
-updatePricesAuthToken(...)

Checklist: Author Self-Review:
 ✅I have performed a self-review of my own code (including running it).
✅ I understand the changes I'm proposing and why they are needed.
✅ My changes generate no new warnings or errors (linting, console).

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test coverage for price-related API utilities.
  * Verifies configuration detection, client initialization with correct base URL and authentication behavior, and return of the created client.
  * Validates token management: updating stored auth token (including null), and correct handling when token or price state is missing.

### CODE REVIEW

File: src/lib/api/prices.test.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
+import { describe, it, expect, vi, beforeEach } from 'vitest';
+type Preferences = Parameters<Parameters<typeof import('$lib/settings').preferences.update>[0]>[0];
```
Comment (VaiTon) [inline src/lib/api/prices.test.ts:2]: What is this?

**Review by VaiTon — CHANGES_REQUESTED** [review]

**Comment by VaiTon** [comment]
Closing as stale.
